client/gutenberg/extensions/slideshow/view.js

```diff
@@ -0,0 +1,23 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSwiper from './create-swiper';
+import './style.scss';
+
+typeof window !== 'undefined' &&
+	window.addEventListener( 'load', function() {
+		const slideshowBlocks = document.getElementsByClassName( 'wp-block-jetpack-slideshow' );
+		forEach( slideshowBlocks, slideshowBlock => {
+			const { effect } = slideshowBlock.dataset;
```

Comment:
No, I think I'm at peace with it now :slightly_smiling_face: 

I think what worried me was the amount of work needed to extract information from the 'serialized' block, but that also included images (which never were stored in `data-` attributes themselves). If it's just a few `data-` attributes that we can gather from the block's top level element, I'm good with it. 

I've come to the conclusion that there probably isn't a better way to strongly associate these data with individual blocks (since, as we discussed on one of our calls or per DM, storing in some `window` global would still need to associate them with the corresponding block via some ID, which would amount to the same complexity the `data-` attribute approach has, and then some more on top of it).